source/client-side-encryption/tests/distinct.template.yml

```diff
@@ -0,0 +1,87 @@
+runOn:
+  - minServerVersion: "4.1.10"
+database_name: &database_name "default"
+collection_name: &collection_name "default"
+
+data:
+  - &doc0_encrypted { _id: 1, ssn: {{ciphertext("457-55-5462", field="ssn")}} }
+  - &doc1_encrypted { _id: 2, ssn: {{ciphertext("457-55-5462", field="ssn")}} }
+  - &doc2_encrypted { _id: 3, ssn: {{ciphertext("123-45-6789", field="ssn")}} }
+json_schema: {{schema()}}
+key_vault_data: [{{key()}}]
+
+tests:
+  - description: "distinct with deterministic encryption"
+    clientOptions:
+      autoEncryptOpts:
+        kmsProviders:
+          aws: {} # Credentials filled in from environment.
+    operations:
+      - name: distinct
+        arguments:
+          fieldName: "ssn"
+          filter: { ssn: "457-55-5462" }
+        result:
+          - "457-55-5462"
+    expectations:
+      # Auto encryption will request the collection info.
+      - command_started_event:
+          command:
+            listCollections: 1
+            cursor: {}
+            filter:
+              name: *collection_name
+          command_name: listCollections
+      - command_started_event:
+          command:
+            listCollections: 1
+            cursor: {}
+            filter:
+              name: "datakeys"
+            $db: admin
+          command_name: listCollections
+      # Then key is fetched from the key vault.
+      - command_started_event:
+          command:
+            find: datakeys
+            filter: { _id: { $in: [ {{key()["_id"]}} ] } }
+            $db: admin
+          command_name: find
+      - command_started_event:
+          command:
+            distinct: *collection_name
+            key: ssn
+            query: { ssn: {$eq: {{ciphertext("457-55-5462", field="ssn")}} } }
+          command_name: distinct
+    outcome:
+      collection:
+        # Outcome is checked using a separate MongoClient without auto encryption.
+        data:
+          - *doc0_encrypted
+          - *doc1_encrypted
+          - *doc2_encrypted
+  - description: "Distinct fails when filtering on a random encrypted field"
```

Comment:
Change to "a deterministic encrypted field".

Reply:
This is actually a duplicate test. Removed.